pkg/api/vlabs/validate.go

```diff
@@ -700,6 +700,9 @@ func (a *Properties) validateAddons() error {
 					if a.HasCoreOS() {
 						return errors.New("flexvolume add-ons not currently supported on coreos distro. Please use Ubuntu")
 					}
+					if common.IsKubernetesVersionGe(a.OrchestratorProfile.OrchestratorVersion, "1.16.0") {
+						log.Warnf("%s add-on will be DEPRECATED in favor of csi-secrets-store addon for 1.16+", addon.Name)
+					}
```

Comment:
Should we add validation that fails if *both* keyvault-flexvol and cse-secrets-store addons are enabled at the same time? Or is that actually a sane configuration?

Reply:
In terms of functionality, both the solutions offer the same and pod configuration to consume the addon is different.  Having both of them run at the same time will not result in any side effects. If we do decide to add the validation then we need to request the user explicitly disable `csi-secrets-store` addon which is default in 1.16+. Do we have similar experience for any other addon in aks-engine today?

Reply:
We don't let users enable both kube-dns and coredn simultaneously, is an example.

Reply:
yes I think validation should fail if both are enabled.

Reply:
cool! I wanted to make sure we don't introduce a new experience just for this add-on. I'll add that validation.